src/variable.cpp

```diff
@@ -4383,6 +4385,67 @@ int Variable::special_function(char *word, char *contents, Tree **tree, Tree **t
   return 1;
 }
 
+/* ----------------------------------------------------------------------
+   process a type label function in formula
+   push result (numeric type) onto tree or arg stack
+   word = interaction type
+   contents = str between parentheses (one type label)
+   return 0 if not a match, 1 if successfully processed
+   customize by adding a label map function:
+     label(typelabel),blabel(typelabel),alabel(typelabel),
+     dlabel(typelabel),ilabel(typelabel)
+------------------------------------------------------------------------- */
+
+int Variable::labelmap_function(char *word, char *contents, Tree **tree,
+                                Tree **treestack, int &ntreestack,
+                                double *argstack, int &nargstack, int ivar)
+{
+  // word not a match to any label map function
+
+  if (strcmp(word,"label") && strcmp(word,"blabel") &&
```

Comment:
This is a unexpected choice for the function name. For a function "label()" I would expect that it returns the label to a numeric type, not the numeric type of a label. So it would be better to call it, for example, "label2type()".

Reply:
ok, makes sense. i think Steve suggested these names, so he may have a stronger opinion

Reply:
@sjplimp how about implementing only one function, label2type(), and then using atom,bond,angle,dihedral,improper as the first argument and the type label as the second, e.g.:
```
variable a1 equal label2type(atom,C1)
variable b1 equal label2type(bond,C1-C1)
```